### PR TITLE
ACSS, PHP & AMSv2 public preview - version 1

### PR DESCRIPTION
**Pull request by gargankit-microsoft**
Please follow the link to find more details on the swagger existing in [azure-rest-api-specs-pr : private repo swagger](https://github.com/Azure/azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/workloads/resource-manager/Microsoft.Workloads). That has already been reviewed by respective teams and is in private preview.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [X] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. **June-July Timeframe**
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. **As early as possible to support further rollouts and SDK Generation**
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

Existing request for SDKs: https://github.com/Azure/sdk-release-request/projects/1164

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by gargankit-microsoft** [comment]
policy check is failing due to 

> The term [whitelist](http://policheck.azurewebsites.net/Pages/TermInfo.aspx?LCID=9&TermID=80411) was used in file 'custom-words.txt' and may not be appropriate to ship in all contexts.  When industry naming conventions or consistency make replacement of the term impractical, leave the term unchanged. When used to describe a list of people or things that have received approval or are considered to be in good favor, replace with "allow list". More specific terms such as "approved list" or "safe recipients list" can be used where appropriate.

Seems unrelated to our changes, as the flagged term is already there. The flag count of three seems to be related to three existing words in file

> whitelisted
> Whitelisting
> Whitelistings

**Comment by gargankit-microsoft** [comment]
policy check is failing due to 

> The term [whitelist](http://policheck.azurewebsites.net/Pages/TermInfo.aspx?LCID=9&TermID=80411) was used in file 'custom-words.txt' and may not be appropriate to ship in all contexts.  When industry naming conventions or consistency make replacement of the term impractical, leave the term unchanged. When used to describe a list of people or things that have received approval or are considered to be in good favor, replace with "allow list". More specific terms such as "approved list" or "safe recipients list" can be used where appropriate.

Seems unrelated to our changes, as the flagged term is already there. The flag count of three seems to be related to three existing words in file

> whitelisted
> Whitelisting
> Whitelistings

**Comment by gargankit-microsoft** [comment]
.NET SDK is failing for some time now for multiple PRs, hence our check is failing for that as well. this is unrelated to change and is caused due to script permissions, which is not in our control.
We are already in touch with owner of .NET SDK to get it generated offline.

File: specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/phpWorkloads.json [hunk 1]
```diff
@@ -0,0 +1,1362 @@
...
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Workloads": {
+            "$ref": "./examples/phpworkloads/WordpressInstances_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Workloads/phpWorkloads/{phpWorkloadName}/wordpressInstances/default": {
```
Comment (NetRock) [inline specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/phpWorkloads.json:368]: Why "default"? is there any other wordpressInstance?

File: specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/phpWorkloads.json [hunk 2]
```diff
@@ -0,0 +1,1362 @@
...
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Workloads": {
+            "$ref": "./examples/phpworkloads/WordpressInstances_Get.json"
+          }
+        }
+      },
+      "put": {
```
Comment (NetRock) [inline specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/phpWorkloads.json:409]: This is t create/update default wordpressInstance?

File: specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/phpWorkloads.json [hunk 3]
```diff
@@ -0,0 +1,1362 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Workloads/phpWorkloads": {
```
Comment (NetRock) [inline specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/phpWorkloads.json:37]: No CRUD on subscription level phpWorkloads?

**Review by NetRock — CHANGES_REQUESTED** [review]

File: specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/SAPVirtualInstance.json [hunk 4]
```diff
@@ -0,0 +1,3324 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Workloads/locations/{location}/sapVirtualInstanceMetadata/default/getSizingRecommendations": {
```
[the remark this answers is not in the document]
Comment (gargankit-microsoft) [reply specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/SAPVirtualInstance.json:37]: No, there are no other values. this is to have a post call with location-based routing.
sapVirtualInstanceMetadata is a ProxyOnly, LocationBased resource.

File: specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/phpWorkloads.json (hunk 1 above)
Comment (venu-sivanadham) [reply specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/phpWorkloads.json:368]: No, we want to restrict number of WordPress instances to 1. Hence using "default" as the only instance.

File: specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/phpWorkloads.json (hunk 3 above)
Comment (gargankit-microsoft) [reply specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/phpWorkloads.json:37]: that would be below API

"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Workloads/phpWorkloads/{phpWorkloadName}"


Note: <b>Please follow the link to find more details on the swagger existing in [azure-rest-api-specs-pr: private repo swagger](https://github.com/Azure/azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/workloads/resource-manager/Microsoft.Workloads). That has already been reviewed by respective teams and is in private preview right now.<b>

File: specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/phpWorkloads.json (hunk 2 above)
Comment (gargankit-microsoft) [reply specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/phpWorkloads.json:409]: This is request contract we have in place to create that "default" instance

**Comment by gargankit-microsoft** [comment]
merged from wrong branch, will raise another one with clear diffs

**Comment by gargankit-microsoft** [comment]
moved to https://github.com/Azure/azure-rest-api-specs/pull/19232 as this was having issues in running checks

**Pull request closed without merging**